src/infra/dex/zeroex/mod.rs

```diff
@@ -168,7 +168,17 @@ pub enum Error {
 impl From<util::http::RoundtripError<dto::Error>> for Error {
     fn from(err: util::http::RoundtripError<dto::Error>) -> Self {
         match err {
-            util::http::RoundtripError::Http(err) => Self::Http(err),
+            util::http::RoundtripError::Http(err) => {
+                if let util::http::Error::Status(code, _) = err {
+                    match code.as_u16() {
+                        429 => Self::RateLimited,
+                        451 => Self::UnavailableForLegalReasons,
```

Comment:
Nit: Why not to match with the code directly (less error prone)?

```suggestion
                 match code {
                        StatusCode::TOO_MANY_REQUESTS => Self::RateLimited,
                        StatusCode::TEMPORARY_REDIRECT => Self::UnavailableForLegalReasons,
```

Reply:
Temporary redirect is 307, though. 😅

Reply:
@MartinquaXD good catch, edited! :ok_hand: